ECLIPs/ECIP-draft_succinct-proofs.md

```diff
@@ -0,0 +1,142 @@
+
+The current data structures in Ethereum (merkle-patricia-trees) were made with the sole purpose of allowing succinct verification of all the data. I have personally been working for some time on the missing tools to extract, send, and validate these proofs. This work is mostly done, but there is a problem: 
+
+Without a succinct way to verify work, there is nothing to verify these merkle proofs _against_.
+
+A merkle proof can only verify against a hash that I _already_ trust. The validation chain goes something like this for some contract data `x`:
+
+- `x` is proven to have existed in a particular `storageRoot_x`
+- `storageRoot_x` is proven to have existed in a particular `account_x`
+- `acount` is proven to have existed in a particular `stateTree_x`
+- `stateTree_x` is proven to have existed in a particular `block_x`
+
+And merkle proofs end there: `block_x` unfortunately cannot be trusted.
+
+There are 2 main strategies for verifying a `block`:
+- Validate every transaction sinse genesis 
+  - This  takes a week on desktop and 
+  - Requires the user to have the full dataset anyway, removing the need for merkle proofs in the first place.
+  - Is infeasible on mobile.
+- Validating the work (of the entire chain)
+  - Relies on majority honest miner assumption
+  - _currently_ requires all block headers
+  - _currently_ requires validating every blockHeader's PoW
+
+<!-- at all unless you check its proof of work. Doing that would yeild a cost parameter:
+
+- `block_x` is proven to have cost about `5 Ether` to produce.
+
+nowing that something cost 5 Ether is not sufficient for most applications. By checking a suffix (blocks after this point in the chain) we can prove that the data cost at least `5n Ether` to produce, where n is the number of blocks checked.
+
+With the simple assumption that the honest chain has 2/3 of the total mining power
+
+The next thing we can do is to request the header before that and validate its PoW as well -->
+
+
+
+The problem is that it ends there. And we still have nothing, if the person (or machine) validating cannot validate the block. The cost to produce such proofs is free; To validate blocks we cant use the same strategy. Blocks validate against a chain, which in turn validates through its proof of work. Currently the only way to do this requires 4gb of chain data. Thats way too much. We need something that can work within a few seconds. 
+
+FlyClient is that something, and adding an `MMRroot` to each blockheader is the protocol change that will allow FlyClient to work.
+
+On a high level, it allows the verifier to sample a subset of blocks in a tree, rather than a full amount thought a linked list, improving the time-complexit from O(n) to O(log(n)). The sample subset can be made large enough such it is cryptographically infeasible to ever produce a false proof.
```

Comment:
```suggestion
On a high level, it allows the verifier to sample a subset of blocks in a tree, rather than a full amount thought a linked list, improving the time-complexity from O(n) to O(log(n)). The sample subset can be made large enough such it is cryptographically infeasible to ever produce a false proof.
```